### PR TITLE
Add windows-amd64 to platforms for CLI builds

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/roadmap/issues/333

This adds building of `windows-amd64` binaries for all CLIs.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by marians** [comment]
Good call for testing. I found out that because the windows file is a ZIP and not .tar.gz, it can't be found.

**Comment by marians** [comment]
The latest commit represents what has been tested with success in https://github.com/giantswarm/kubectl-gs/runs/6027207195?check_suite_focus=true

**Review by anvddriesch — APPROVED** [review]

**Comment by marians** [comment]
Looks like code signing doesn't work as expected yet.

**Review by anvddriesch — APPROVED** [review]
Looks good! Let's run it in `kubectl-gs` first to see if it's working as expected and then merge this.

**Comment by marians** [comment]
In kubectl-gs we now have moved the shell code from the Makefile into a [separate script](https://github.com/giantswarm/kubectl-gs/blob/main/.github/windows-code-signing.sh) in the `.github` folder. How can we use devctl to create/maintain that file?

**Comment by ubergesundheit** [comment]
Here is an example PR which shows how you can add files to devctl https://github.com/giantswarm/devctl/pull/379

File: pkg/gen/input/makefile/internal/file/windows-code-signing.sh.template [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+{{ .Header }}
+
```
Comment (ubergesundheit) [inline pkg/gen/input/makefile/internal/file/windows-code-signing.sh.template:4]: Unsure if this script would benefit from `set -e` (http://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#The-Set-Builtin) to make the script exit immediately on any error.

Unsure if this exist in plain /bin/sh

**Review by ubergesundheit — APPROVED** [review]
LGTM if tested

**Comment by marians** [comment]
For testing, I am executing it like this:

```
go run main.go gen makefile -f cli -l go
```

however I don't get a `windows-code-signing.sh` file as a result.

**Comment by ubergesundheit** [comment]
Check pkg/gen/gen.go line 52 [link](https://github.com/giantswarm/devctl/blob/master/pkg/gen/gen.go#L52=)

The function returns false and nothing gets done.

So either you change the file name to start with `"zz_generated."` or you add an exception in the function `isRegenerable` (same file)

**Comment by marians** [comment]
Thank you @ubergesundheit ! As always, that was super helpful!

**Pull request merged**